src/packagedcode/pypi.py

```diff
@@ -192,6 +194,10 @@ def parse_with_dparse(location):
     if is_dir:
         return
     file_name = fileutils.file_name(location)
+    
+    if file_name.endswith('requirements.txt') or file_name.endswith('requirements.in'):
```

Comment:
endswith() takes a tuple and does not need to be called twice. Also what about `requirements-dev.txt` and similar?
IMHO you should make this a separate function and you can then have a series of unit test for it.